testutil/keeper/unit_test_helpers.go

```diff
@@ -22,169 +23,173 @@ import (
 	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
 	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
 	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v3/modules/core/23-commitment/types"
+	ibctmtypes "github.com/cosmos/ibc-go/v3/modules/light-clients/07-tendermint/types"
+	providertypes "github.com/cosmos/interchain-security/x/ccv/provider/types"
 )
 
-// Constructs a provider keeper and context object for unit tests, backed by an in-memory db.
-func GetProviderKeeperAndCtx(t testing.TB) (providerkeeper.Keeper, sdk.Context) {
+// Parameters needed to instantiate an in-memory keeper
+type InMemKeeperParams struct {
+	Cdc            *codec.ProtoCodec
+	StoreKey       *storetypes.KVStoreKey
+	ParamsSubspace *paramstypes.Subspace
+	Ctx            sdk.Context
+}
 
-	cdc, storeKey, paramsSubspace, ctx := SetupInMemKeeper(t)
+// NewInMemKeeperParams instantiates in-memory keeper params with default values
+func NewInMemKeeperParams(t testing.TB) InMemKeeperParams {
+	storeKey := sdk.NewKVStoreKey(types.StoreKey)
+	memStoreKey := storetypes.NewMemoryStoreKey(types.MemStoreKey)
+
+	db := tmdb.NewMemDB()
+	stateStore := store.NewCommitMultiStore(db)
```

Comment:
Here a new store is created, so each paramspace returned (in successive calls to this function) will have a different underlying store object. Does that work out ok?

I like it.

Reply:
Interesting question, I don't think there's a use case for creating multiple keepers instantiations in a single unit test (that would exist at the same time). The assumption here is that each test case will make a new lightweight, in-mem keeper that will be cleaned up by the garbage collector at the end of the test case. This pattern has worked thus far, but I can't claim that I know every intricacy of how the in-mem keeper works. This design was initially based off the [cosmos developer tutorial source code](https://github.com/cosmos/b9-checkers-academy-draft/blob/b79a43c6e1db1c1df42290fbd037fa87192902a3/x/checkers/keeper/keeper_test.go).

Reply:
Ok we can change later if needed 👍